src/plugins/data_sources/server/plugin.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+} from '../../../core/server';
+
+import { defineRoutes } from './routes';
+import { dataSource } from './saved_objects';
+import { DataSourcePluginSetup, DataSourcePluginStart } from './types';
+
+export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourcePluginStart> {
+  private readonly logger: Logger;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+  }
+
+  public setup(core: CoreSetup) {
+    core.savedObjects.registerType(dataSource);
+
+    this.logger.debug('dataSourceManagement: Setup');
+    const router = core.http.createRouter();
+
+    // Register server side APIs
+    defineRoutes(router);
```

Comment:
we may not need new routes for data source operations as it has been taken care by saved objects service

Reply:
However, we may need to define data source model on server side, so that other component, like opensearch service, can use it